ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/TransactionGasCalculator.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright contributors to Hyperledger Besu
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.hyperledger.besu.ethereum.mainnet;
+
+import org.hyperledger.besu.ethereum.core.Gas;
+import org.hyperledger.besu.ethereum.core.GasAndAccessedState;
+import org.hyperledger.besu.ethereum.core.Transaction;
+
+/**
+ * Provides various gas cost lookups and calculations used during block processing.
+ *
+ * <p>The {@code GasCalculator} is meant to encapsulate all {@link Gas}-related calculations except
+ * for the following "safe" operations:
+ *
+ * <ul>
+ *   <li><b>Operation Gas Deductions:</b> Deducting the operation's gas cost from the VM's current
+ *       message frame because the
```

Comment:
finish sentence

Reply:
That incomplete sentence dates back to Pantheon code.  But because I touched it I get to fix it ;)

Reply:
It's only fair.

Reply:
done